core/services/ocr2/plugins/ocr2keeper/evm21/transmit_event_provider.go

```diff
@@ -141,7 +141,136 @@ func (c *LogProvider) HealthReport() map[string]error {
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to get latest block from log poller", err)
+	}
+
+	// always check the last lookback number of blocks and rebroadcast
+	// this allows the plugin to make decisions based on event confirmations
+	logs, err := c.logPoller.LogsWithSigs(
+		end-c.lookbackBlocks,
+		end,
+		[]common.Hash{
+			iregistry21.IKeeperRegistryMasterUpkeepPerformed{}.Topic(),
+		},
+		c.registryAddress,
+		pg.WithParentCtx(ctx),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to collect logs from log poller", err)
+	}
+
+	performed, err := c.unmarshalPerformLogs(logs)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to unmarshal logs", err)
+	}
+
+	vals := []ocr2keepers.TransmitEvent{}
+
+PerformLoop:
+	for _, p := range performed {
+		// abi decode trigger bytes into `wrappedTrigger`
+
+		var Uint32 = mustNewType("uint32", "", nil)
+		var Hash = mustNewType("bytes32", "", nil)
+		args := abi.Arguments{
+			{Name: mKeys[0], Type: Hash},
+			{Name: mKeys[1], Type: Uint32},
+			{Name: mKeys[2], Type: Uint32},
+			{Name: mKeys[3], Type: Hash},
+		}
+
+		/*
+		  TODO: two different structs exist in the contract. we are assuming the
+		  the LogTrigger struct here, but it could be the BlockTrigger as well
+		  struct BlockTrigger {
+		    uint32 blockNum; // TODO - only 34 years worth of blocks on arbitrum...
+		    bytes32 blockHash;
+		  }
+
+		  struct LogTrigger {
+		    bytes32 txHash;
+		    uint32 logIndex;
+		    uint32 blockNum;
+		    bytes32 blockHash;
+		  }
+		*/
+
+		keys := []string{"txHash", "logIndex", "blockNum", "blockHash"}
+		values := make(map[string]interface{})
+		if err := args.UnpackIntoMap(values, p.Trigger); err != nil {
+			c.logger.Error("error unpacking trigger values: %w", err)
+			continue PerformLoop
+		}
+
+		for _, key := range keys {
+			if _, ok := values[key]; !ok {
+				c.logger.Error("error unpacking trigger values: %s missing from struct", key)
+				continue PerformLoop
+			}
+		}
+
+		tHsh, ok := values[keys[0]].([32]byte)
+		if !ok {
+			c.logger.Error("error unpacking trigger values: %s is incorrect type", keys[0])
+			continue PerformLoop
+		}
+
+		lIdx, ok := values[keys[1]].(uint32)
+		if !ok {
+			c.logger.Error("error unpacking trigger values: %s is incorrect type", keys[1])
+			continue PerformLoop
+		}
+
+		bln, ok := values[keys[2]].(uint32)
+		if !ok {
+			c.logger.Error("error unpacking trigger values: %s is incorrect type", keys[2])
+			continue PerformLoop
+		}
+
+		bHsh, ok := values[keys[3]].([32]byte)
+		if !ok {
+			c.logger.Error("error unpacking trigger values: %s is incorrect type", keys[3])
+			continue PerformLoop
+		}
+
+		logExtension := fmt.Sprintf("%s:%d", common.BytesToHash(tHsh[:]).Hex(), uint(lIdx))
+		trigger := ocr2keepers.NewTrigger(int64(bln), common.BytesToHash(bHsh[:]).Hex(), logExtension)
+		payload := ocr2keepers.NewUpkeepPayload(
+			p.Id,
+			int(logTrigger),
+			ocr2keepers.BlockKey(fmt.Sprintf("%d", trigger.BlockNumber)),
```

Comment:
this block number from trigger not the check block. wonder if this matters

Reply:
It won't matter at the moment because the block key isn't used to create the payload id.